File: benchmarks/pandas/bench_rolling_std.py
"""Benchmark: Series.rolling().std() — rolling standard deviation."""
import json, time
import pandas as pd

SIZE = 100_000
WINDOW = 20
WARMUP = 5
ITERATIONS = 50

s = pd.Series([float(i*1.1+0.5) for i in range(SIZE)])

for _ in range(WARMUP):
    s.rolling(WINDOW).std()

times = []
for _ in range(ITERATIONS):
    t0 = time.perf_counter()
    s.rolling(WINDOW).std()
    times.append((time.perf_counter() - t0) * 1000)

total_ms = sum(times)
print(json.dumps({"function":"rolling_std","mean_ms":round(total_ms/ITERATIONS,3),"iterations":ITERATIONS,"total_ms":round(total_ms,3)}))

File: benchmarks/tsb/bench_rolling_std.ts
/**
 * Benchmark: Series.rolling().std() — rolling standard deviation.
 * Outputs JSON: {"function": "rolling_std", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series } from "../../src/index.ts";

const SIZE = 100_000;
const WINDOW = 20;
const WARMUP = 5;
const ITERATIONS = 50;

const s = new Series({ data: Array.from({ length: SIZE }, (_, i) => i * 1.1 + 0.5) });

for (let i = 0; i < WARMUP; i++) {
  s.rolling(WINDOW).std();
}

const times: number[] = [];
for (let i = 0; i < ITERATIONS; i++) {
  const start = performance.now();
  s.rolling(WINDOW).std();
  times.push(performance.now() - start);
}

const totalMs = times.reduce((a, b) => a + b, 0);
const meanMs = totalMs / ITERATIONS;
console.log(
  JSON.stringify({
    function: "rolling_std",
    mean_ms: Math.round(meanMs * 1000) / 1000,
    iterations: ITERATIONS,
    total_ms: Math.round(totalMs * 1000) / 1000,
  }),
);
